### ISSUE TITLE
Button not displayed correctly

### ISSUE THREAD

**Issue by funkenstrahlen** [podlove/podlove-subscribe-button#60]
I use the Podlove Publisher on my site and included the subcribe button widget into my sidebar:

![screen shot 2015-05-22 at 15 59 14](https://cloud.githubusercontent.com/assets/144173/7771995/914c1bca-009b-11e5-9cae-6522154d214a.png)

This works fine if I access he page on my desktop computer. See http://podcast.funkenstrahlen.de for reference.

However on iPad and iPhone the button is kind of broken. Still works but looks bad. Any ideas what's wrong?

![fullsizerender](https://cloud.githubusercontent.com/assets/144173/7771954/62236c22-009b-11e5-8ca8-3614fad4eb0a.jpg)

**Comment by JoeyTheHobbes** [comment]
I experience the same issue for my podcast. On iOS 8.3 (iPhone 6), I do get the same result as showed in the initial post.
It does look fine on a Samsung Galaxy S3 (Android) though.
I checked the same thing on a Windows Phone too: There, the lgo and the button are not even displayed.

**Comment by benzimmer** [comment]
@Inrumpo Could you give me a link to your podcast website please. I have been trying to replicate the behaviour and did not succeed so far. It seems to have something to do with the button not being visible on page load, but as already said, I cannot replicate the issue consistently.

**Comment by jonasschoen** [comment]
Their podcast websites:

- [Funkenstrahlen](http://podcast.funkenstrahlen.de/)
- [GameTalk](http://gametalk.fm/)

They both use the [Twenty Fifeteen](https://wordpress.org/themes/twentyfifteen/) theme. I also used this for my podcast [Technische Aufklärung](http://technische-aufklaerung.de) and experienced the same problem. I switched to Twenty Thirteen now and the problem is gone. So I think it has something to do with the theme.

**Comment by funkenstrahlen** [comment]
Any news on this? Will it be fixed? I think the subscribe button should not interfere with themes. At least not with official wordpress themes.

**Comment by benzimmer** [comment]
So, I finally figured out where the bug comes from :)

It only happens when the button is set to adapt to the width of the container it is in. I will try to fix this, but I think it will need a major rewiring, so it won't happen any time soon (I have created a separate issue for this here https://github.com/podlove/podlove-subscribe-button/issues/72). As a workaround you can deactivate the 'Autowidth' option in the plugin (or the widget configuration). However, there currently is a bug in the Wordpress plugin that always tells the button to use the autowidth option (https://github.com/podlove/podlove-subscribe-button-wp-plugin/issues/7). I hope this will be fixed soon.

**Comment by benzimmer** [comment]
This seems to be gone with the new version of the button. Please reopen if you still encounter any issues.

**Comment by funkenstrahlen** [comment]
I can confirm the issue I reported above is gone with the new version! Thank you! :)

However I see an issue with Safari on another page: https://podperlen.de/abonnieren/

If I press CMD-R on this page, the button renders like this:
![screen shot 2016-04-28 at 10 40 30](https://cloud.githubusercontent.com/assets/144173/14880241/118b782c-0d2e-11e6-8cd0-18f8c763ae48.png)

If I access the page via the sidebar menu (Abonnieren) it renders fine.

Does not occur in Chrome.

**Comment by benzimmer** [comment]
Can you give me the Safari version you are using. I can't reproduce the problem with 9.1 (11601.5.17.1).

**Comment by funkenstrahlen** [comment]
Interesting. I use the exact same version: Version 9.1 (11601.5.17.1)

**Comment by funkenstrahlen** [comment]
Seems to be a bug with Adblockers. After disabling Ghostery and Adblock, it works fine. I see this in my console:

![screen shot 2016-04-28 at 10 53 39](https://cloud.githubusercontent.com/assets/144173/14880528/8c827e08-0d2f-11e6-8551-4f3bf77f9908.png)

I use both blockers in Chrome too though.

**Comment by funkenstrahlen** [comment]
Now I see that it is not 100% reproducable. After Refreshing CMD-R some time and disabled adblockers the issue and error is visible again.

**Comment by benzimmer** [comment]
Strange. Just installed both blockers. Reloaded like 20 times and never saw the error...

The ghostery blocking message does not seem to affect the button at all from what I can see.

**Comment by funkenstrahlen** [comment]
It even switches between different sizes on reloads

![screen shot 2016-04-28 at 11 02 16](https://cloud.githubusercontent.com/assets/144173/14880720/b0456ac0-0d30-11e6-9176-668c0b8a1aef.png)

I try to figure it out somehow what's the problem...

**Comment by funkenstrahlen** [comment]
I now completly removed both extension. The error in the log is gone. But the problem is still there.

![screen shot 2016-04-28 at 11 05 36](https://cloud.githubusercontent.com/assets/144173/14880800/2b78f586-0d31-11e6-83c3-8babfa802e0b.png)

**Comment by funkenstrahlen** [comment]
Looks like a very weird Safari Bug in my eyes.

**Comment by funkenstrahlen** [comment]
This could be a timing issue as the sizes render differently on reloads? They grow when I have the debugger enabled.

See this little video for demonstration:

http://quick.as/X4bGHBmB3

**Comment by funkenstrahlen** [comment]
Despite this weird thing: I like the new button very much! It looks great :)

**Comment by funkenstrahlen** [comment]
I can not reproduce this bug on other pages. So what's different on my page...

**Comment by benzimmer** [comment]
When set to `auto` the button width depends on the size of the surrounding element when it is rendered. So, yes, this could be a timing issue.

As a workaround you could try to add sth. like this (not really tested) to the bottom of the page source:

```
<style>
  .podlove-subscribe-button-iframe {width: 300px !important;}
</style>
```

I'll reopen this and see if we can find a permanent solution for this.

**Comment by funkenstrahlen** [comment]
Probably Safari calculates the width of the surrounding element in parallel to embedding the iframe. Chrome seems to do that better somehow... #OnlyApple

**Comment by funkenstrahlen** [comment]
Setting the width **and** height of the iframe kinda fixed it :)

![screen shot 2016-04-28 at 11 32 57](https://cloud.githubusercontent.com/assets/144173/14881610/0db87e32-0d35-11e6-873d-88ce7b73d354.png)

I also tried setting the containing element width to a fixed width. However setting the width of the containing `div` to fixed 300px did not help:

```html
<style>
  div.page {width: 300px !important;}
</style>
```

**Comment by funkenstrahlen** [comment]
You still can not reproduce this on your machine?

**Comment by eteubert** [comment]
I can confirm the render bug in Safari and that setting a fixed iframe height in CSS helps.

**Comment by benzimmer** [comment]
Crazy that I can't reproduce it. Maybe we can have a look on the weekend together @eteubert

**Comment by Fischaela** [comment]
Is this still an issue?

**Comment by saerdnaer** [comment]
@funkenstrahlen Can we close this issue?
